src/code/sys_ucode.c

```diff
@@ -1,20 +1,20 @@
 #include "global.h"
 
-u32 D_8012DBA0 = (u32)&D_80155F50;
-u32 D_8012DBA4 = (u32)&D_80157580;
+u64* sDefaultGSPUcodeText = gspF3DZEX2_NoN_PosLight_fifoTextStart;
+u64* sDefaultGSPUcodeData = gspF3DZEX2_NoN_PosLight_fifoDataStart;
 
-u32 SysUcode_GetUCodeBoot(void) {
-    return &D_80009320;
+u64* SysUcode_GetUCodeBoot(void) {
+    return rspbootTextStart;
 }
 
-u32 SysUcode_GetUCodeBootSize(void) {
-    return (u32)&D_800093F0 - (u32)&D_80009320;
+size_t SysUcode_GetUCodeBootSize(void) {
+    return (size_t)((u8*)&rspbootTextEnd - (u8*)&rspbootTextStart);
```

Comment:
I think a `uintptr_t` may be a better type for this cast instead of a `u8*`, but I'm not completely sure. Any reason to prefer one over the other?

Reply:
As much as possible I prefer to stick to pointers rather than moving straight to uintptr_t but maybe this is misguided? I'm not really sure.